### PR TITLE
Migrate `components/picker`

### PR DESCRIPTION
**Pull request by Etheryte**
## What does this PR change?

Migrate all components in `components/picker` to Typescript.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Part of https://github.com/SUSE/spacewalk/issues/13740

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by hustodemon — APPROVED** [review]
LGTM!

File: web/html/src/global.d.ts [hunk 1]
```diff
@@ -47,6 +47,9 @@ declare global {
 
   // Used by section-toolbar, defined in spacewalk-essentials.js
   var handleSst: Function;
+
+  /** DEPRECATED: Do **NOT** use this global for new code, prefer `useUserLocalization()` instead */
```
Comment (ncounter) [inline web/html/src/global.d.ts:51]: :+1:

File: web/html/src/global.d.ts [hunk 2]
```diff
@@ -47,6 +47,9 @@ declare global {
...
+  var localTime: string | undefined;
 }
 
 export {};
```
Comment (ncounter) [inline web/html/src/global.d.ts:55]: Not part of this PR but, could you please add a final empty line? Git/GitHub complain about it, thanks.

File: web/html/src/components/picker/recurring-event-picker.tsx [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+import { DateTimePicker } from "../datetimepicker";
+import { Combobox } from "../combobox";
+import { ComboboxItem } from "../combobox";
+import { Form } from "components/input/Form";
+import { Text } from "components/input/Text";
+import { Utils } from "utils/functions";
+import styles from "./recurring-event-picker.css";
+
+type RecurringType = "hourly" | "daily" | "weekly" | "monthly" | "cron";
+// TODO: This should be `Record<string, string>` or `Record<string, number>`, but currently they're used mixed up
```
Comment (ncounter) [inline web/html/src/components/picker/recurring-event-picker.tsx:12]: I didn't took a deep look into this: how hard/long could it be to address and fix this? Would it be worth of doing it right now or we can live with it?

**Review by ncounter — APPROVED** [review]
I left a couple of comments but more as questions. The PR LGTM :+1:

File: web/html/src/components/picker/recurring-event-picker.tsx (hunk 3 above)
Comment (Etheryte) [reply web/html/src/components/picker/recurring-event-picker.tsx:12]: This would require functional changes, something that I'm trying to avoid in the migration since it would require considerably deeper testing of everything related. Essentially the problem is that some places set the values as strings while others set it as numbers and it isn't trivial to refactor this without testing it through the full stack.  

Since this code works right now, I don't see a need to put this time and effort into here right now, but this TODO is a good option for anyone who happens to work more in depth on this piece in the future.
